### PR TITLE
Automate linting, custom data types

### PR DESCRIPTION
**Pull request by CBroz1**
In this PR, I... 

- [X] Revise powers and bestiary to use dataclasses. This concentrates assumptions about each data structure into the relevant class and, I hope, makes edits more managable.
- [X] Re-implement export support for markdown and csv from yaml spec files
- [X] Remove support for dot files (i.e., flow chart images of powers). This is archived on my local machine. It does not work on the current data class structure, but could be re-added given a strong need. Mermaid may be a better choice if we wanted flowcharts.
- [X] Add full bestiary support 
  - #151 
  - #64 
- [X] Add html templating and html -> png support for PCs. Example below. #83 
- [X] Add support for a StatOverride Power attribute that revises the character stats who take it. #160 This falls short of full implementation, as this PR does not add the yaml equivalent of each vulnerability
- [X] Previously fixed by other PR: #105
- [X] Adding approx 5 #107 
- [X] Rest cards = HP, Quick rest draw 10+2*VIT #156 
- [X] Adding csv. Not dash table actively filterable per #5 but fulfills actual need
- [X] Changing Fatigue structure #152

### CODE REVIEW

File: README.md
Comment (LockerM) [inline README.md:3]:
```suggestion
This repository establishes the version history of the Deck of Adventures tabletop
```

File: README.md
Comment (LockerM) [inline README.md:5]:
```suggestion
Adventures, it’s your turn, your choice.
```

File: README.md
Comment (LockerM) [inline README.md:3]:
```suggestion
This repository establishes the version history of the Deck of Adventures tabletop
```

File: automation/_input/04_Powers_SAMPLE.yaml
Comment (LockerM) [inline automation/_input/04_Powers_SAMPLE.yaml:13]: Does Stat Override reduce the existing stat by the integer or does it auto-apply that number as a static? i.e. at char creation I take Clumsy -1 to my AGL so my AGL is -1, but then at next level I bump AGL by 1 to get my AGL to 0 - will the Clumsy Power continue to overwrite that change or does it reduce by this int?

File: automation/_input/04_Powers_SAMPLE.yaml
Comment (LockerM) [inline automation/_input/04_Powers_SAMPLE.yaml:35]: Do we want to create a -1 minor and -2 major for each Attribute? I can work on that using this as a template.

File: automation/_input/06_Bestiary_SAMPLE.yaml
Comment (LockerM) [inline automation/_input/06_Bestiary_SAMPLE.yaml:63]: If we use this as a template and create additional PCs using this guide, would this be able to generate N character sheets in the new format reading for "type: PC"?

**Review by LockerM — COMMENTED** [review]
Minor changes and a few comments, but overall looks really good! Will be ready to merge once minor text fixes are updated and question on Stat Override is resolved (doesn't need to be solved before merge but discussion of question in general).

File: automation/_input/04_Powers_SAMPLE.yaml
Comment (CBroz1) [reply automation/_input/04_Powers_SAMPLE.yaml:13]: Current implementation replaces the value [here](https://github.com/CBroz1/TheGame/blob/main/automation/templates/bestiary.py#L250), but I can make edits to change to a sum

File: automation/_input/04_Powers_SAMPLE.yaml
Comment (CBroz1) [reply automation/_input/04_Powers_SAMPLE.yaml:35]: An alternate implementation I'd have to think about is using the Options and Choice parameters, but that would take a little more work on my part. Meaning one Attrib vulny, but choosing the attrib on selection. The easier thing would be to copy/paste the above 6 times and make edits accordingly

File: automation/_input/06_Bestiary_SAMPLE.yaml
Comment (CBroz1) [reply automation/_input/06_Bestiary_SAMPLE.yaml:63]: Yes with the following caveat - Currently, they will all have the same inventory (unless passed as a dict to the `items` param), and the inventory will have no impact on stats.

**Review by LockerM — APPROVED** [review]
Updated changes and edits look good. Ready to merge from my end.

**Pull request merged**
